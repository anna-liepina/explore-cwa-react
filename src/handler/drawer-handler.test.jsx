import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe.skip('<TreeHandler/>', () => {
    const data = [
        {
            text: 'root',
            nodes: [
                {
                    text: 'children',
                }
            ]
        },
    ];
    const props = {
        onFilter: jest.fn(),
        onExpand: jest.fn(),
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<TreeHandler {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['title', '{{title}}'],
                ['data-cy', '{{data-cy}}'],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<TreeHandler {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });

            it('if node has isVisible property as false, it should NOT render it', () => {
                const c = shallow(<TreeHandler {...props} data={data.map((v) => ({ ...v, isVisible: false }))} />);

                expect(c).toMatchSnapshot();
            });
        });
    });

    describe('internal callbacks', () => {

    });
});
