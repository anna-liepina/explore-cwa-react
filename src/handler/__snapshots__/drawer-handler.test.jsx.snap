// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerHandler/> render with default/required props 1`] = `
<DocumentFragment>
  <button
    class=""
    data-cy="--drawer--open"
  />
</DocumentFragment>
`;

exports[`<DrawerHandler/> render with optional/required props 1`] = `
<DocumentFragment>
  <button
    class="{{className}}"
    data-cy="{{data-cy}}--drawer--open"
  />
</DocumentFragment>
`;
